### PR TITLE
Python acceptance tests

### PR DESCRIPTION
**Pull request by dirkkul**
### What's being changed:

Adds framework for python acceptance test

This:
- allows to run python tests from the standard run.sh script
  - can also be used directly using `pytest -n auto *path*`
- Adds an extra CI job
- automatically creates a venv with the correct dependencies and activates it

All tests are run in parallel by default and clean up before and after their execution

### CODE REVIEW

File: test/acceptance_with_python/run.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# Activate the virtual environment
+source .venv/bin/activate
+
+cd "$SCRIPT_DIR" || return
+
+pip install --upgrade pip --quiet
+pip install -r requirements.txt --quiet
+
+# run python tests in parallel
+pytest -n auto .
```
Comment (antas-marcin) [inline test/acceptance_with_python/run.sh:26]: please add a new empty line at the end of the script

File: test/acceptance_with_python/run.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
```
Comment (antas-marcin) [inline test/acceptance_with_python/run.sh:1]: would be great to change this part to:
```
#!/usr/bin/env bash

set -euo pipefail
```
Please set this second setting to be sure that the exit codes will be propagated properly

File: test/run.sh [hunk 3]
```diff
@@ -111,6 +114,13 @@ function main() {
     fi
   fi
 
+  if $run_acceptance_only_python|| $run_all_tests
```
Comment (antas-marcin) [inline test/run.sh:117]: add a space before `||`
```
$run_acceptance_only_python || $run_all_tests
```

File: test/acceptance_with_python/run.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    python3 -m venv .venv
+fi
+
+# Activate the virtual environment
+source .venv/bin/activate
+
+cd "$SCRIPT_DIR" || return
+
+pip install --upgrade pip --quiet
+pip install -r requirements.txt --quiet
```
Comment (antas-marcin) [inline test/acceptance_with_python/run.sh:23]: can we change this to `pip3`? or it doesn't matter? (I have always used `pip3`)

Comment (dirkkul) [reply test/acceptance_with_python/run.sh:23]: it doesn't matter because we activated the virtual env :)

**Review by antas-marcin — APPROVED** [review]

**Pull request merged**